src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void getOperation(List<FruitTransaction> transactions) {
+
+        // someone has bought some fruits
+        List<FruitTransaction> transactionsSelected = transactions.stream()
+                .filter(a -> a.getOperation() == FruitTransaction.Operation.PURCHASE)
+                .toList();
+
+        for (FruitTransaction fruitTransaction : transactionsSelected) {
+            int newValue = Storage.fruits.get(fruitTransaction.getFruit())
+                    - fruitTransaction.getQuantity();
+            Storage.fruits.put(fruitTransaction.getFruit(), newValue);
+        }

```

Comment:
The current implementation directly accesses the `Storage.fruits` map. It is better to encapsulate the access to the storage within the Storage class itself. Consider adding a method in the Storage class to handle the update operation, which would also include the balance check.

Reply:
I'm not sure how to handle Storage.fruits when making it private..